### PR TITLE
7331 (frontend) shared logos

### PR DESCRIPTION
**Pull request by gmarinov**
https://jira.dev.bbc.co.uk/browse/PROGRAMMES-7331

### CODE REVIEW

File: config/services.yaml [hunk 1]
```diff
@@ -84,6 +87,17 @@ services:
     # add more service definitions when explicit configuration is needed
     # please note that last definitions always *replace* previous ones
 
+    # @todo remove _int_ from manifest basepath
```
Comment (ColinBarnwell) [inline config/services.yaml:90]: Looks like you can remove this todo now.

File: src/Branding/LogoVersionStrategy.php [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+            $json = $this->cache->get($cacheKey, function (ItemInterface $item) use ($path, $forceRefresh) {
+                $ttl = $this->cacheTTL;
+                $item->expiresAfter($ttl);
+                $mt = microtime(true);
+                if (!($this->cacheCleared || $forceRefresh)) {
+                    $mt = floor($mt / $ttl) * $ttl;
+                }
+                $path .= is_null(parse_url($path, PHP_URL_QUERY)) ? "?_t_={$mt}" : "&_t_={$mt}";
+                // $this->logger->notice("recomputing cache {$item->getKey()} (ttl: {$ttl}) from path {$path}");
```
Comment (ColinBarnwell) [inline src/Branding/LogoVersionStrategy.php:116]: Another cleanup item.

**Review by ColinBarnwell — APPROVED** [review]
Approval based on walkthrough and my best efforts at following the code, but a lack of familiarity with the framework, coding style and overall codebase means this review is pretty superficial!

File: src/Branding/LogoConfig.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        $self->logos = $obj['logos'];
+        $self->manifest = $obj['manifest'];
+        return $self;
+    }
+
+    /**
+     * Get a unittest fixture
+     * @return LogoConfig
+     */
+    public static function unittest(): self
```
Comment (ajroberts10) [inline src/Branding/LogoConfig.php:57]: are we ok having test specific code in a production file?

Comment (gmarinov) [reply src/Branding/LogoConfig.php:57]: I investigated a few other ways than this factory method to set this up for tests, but it quickly becomes hairy and hard to follow across even more config files. I had partial success overriding those with test-specific versions (which also gives you other problems). More specifically it begins failing in tests that call the site to render content. 🤷‍♂️

File: src/Branding/LogoVersionStrategy.php [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    private function url(string $path): string
+    {
+        return sprintf(self::FORMAT, $this->s3Base, $path);
+    }
+
+    private function loadLogoConfig($forceRefresh = false): LogoConfig
+    {
+        $cosmosInfo = $this->cosmosInfo;
+        if ($cosmosInfo->getAppEnvironment() === 'unittest') {
+            return LogoConfig::unittest();
```
Comment (llh1) [inline src/Branding/LogoVersionStrategy.php:96]: It's a bit invasive to have a mock configuration in the production code. Is there no other way in PHP to mock a function?

File: src/Branding/LogoVersionStrategy.php [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+use App\ValueObject\CosmosInfo;
+use GuzzleHttp\Client;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Asset\VersionStrategy\VersionStrategyInterface;
+use Symfony\Component\Cache\Adapter\AbstractAdapter;
+use Symfony\Contracts\Cache\ItemInterface;
+use App\Branding\LogoConfig;
+
+class LogoVersionStrategy implements VersionStrategyInterface
```
Comment (llh1) [inline src/Branding/LogoVersionStrategy.php:13]: Should we have a unit test for this class? Is it tested somewhere?

File: src/Branding/LogoVersionStrategy.php (hunk 4 above)
Comment (gmarinov) [reply src/Branding/LogoVersionStrategy.php:96]: I did try overriding this class for tests alone and Symfony decided not to play ball - I’ll try again but the spaghetti factor shoots up. example:

Screenshot 2020-11-05 at 12 06 08

File: src/Branding/LogoVersionStrategy.php (hunk 5 above)
Comment (gmarinov) [reply src/Branding/LogoVersionStrategy.php:13]: I’ll port over the test from Brandingtool

**Review by ajroberts10 — APPROVED** [review]
Looks good 👍  nice job

**Review by llh1 — APPROVED** [review]
👍
